bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -459,25 +459,39 @@ object BagsManager {
       }
     }
 
-    obs.getProgram.removeStructureChangeListener(StructureListener)
-    obs.getProgram.removeCompositeChangeListener(ChangeListener)
-    applySelection(TpeContext(obs))
-
-    // Update the TPE if it is visible
-    selOpt.foreach { sel =>
-      Option(TpeManager.get()).filter(_.isVisible).foreach { tpe =>
-        sel.assignments.foreach { ass =>
-          val clazz = ass.guideProbe.getType match {
-            case GuideProbe.Type.AOWFS => classOf[Altair_WFS_Feature]
-            case GuideProbe.Type.OIWFS => classOf[OIWFS_Feature]
-            case GuideProbe.Type.PWFS  => classOf[TpePWFSFeature]
+    val ctx = TpeContext(obs)
+
+    // If the PA is not allowed to change by the PA constraint, but does, then this is an intermediate
+    // result that must be ignored: further changes to the PA have been made since this result was obtained,
+    // so do not process this case.
+    val process = (for {
+      paCst <- ctx.obsContext.map(_.getPosAngleConstraint)
+      oldPA <- ctx.instrument.dataObject.map(_.getPosAngleDegrees)
+      newPA <- selOpt.map(_.posAngle.toDegrees)
+    } yield paCst != PosAngleConstraint.FIXED || oldPA == newPA) | false
```

Comment:
"Smart" is probably a stretch... :-)  At any rate, I think there will be a race condition here unless we change the definition of `RunningEdited.success`.  If we might decide in `applyAction` to skip the update and yet we record the `hash` value before transitioning to `Pending`, we could be in for trouble if the observation is subsequently edited in such a way as to produce the same `hash` value that we just ignored before the pending state wake up is processed.  It pains me, but I guess the safe thing to do then, in addition, would be to change `RunningEdited.success` to just always re-trigger like so:

````scala
override def succeed(results: Option[AgsStrategy.Selection]): StateTransition =
    (PendingState(obs, None), BagsManager.wakeUpAction(obs, 0))
````